### PR TITLE
Added popup modal with transition for redirect when account exists

### PR DESCRIPTION
**Pull request by joeelia**
# Issue Being Addressed

#268 

# Type of PR

Put an `x` in the brackets for the type(s) of PR this is. You may tick more than one.

[ ] Bug Fix
[ ] Refactor
[ ] New Feature
[x] Update to Existing Feature
[ ] Other (state below)

# Description
Error was shown when someone tried to order who already had an account. This notifies a user that they already have an account and gives them the option to go to login page.

# How to Test/Reproduce

Provide steps on how one can test your changes here. e.g.
1. Go to https://deploy-preview-322--foodouken.netlify.app/shop/foodie-27/your-address
2. Select an item and start the order. Fill in any name
3. For email use an existing account **bamboo.chow@gmail.com**
4. Fill in any phone number
5. Notice that you get a popup that lets you go to login page

# Screenshots/casts

Screen Shot 2020-07-10 at 3 24 15 AM

# Additional Context

I think having a popup is better than just automatically redirecting them

### CODE REVIEW

File: src/components/inputs/DuplicateAccount.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  name: 'DuplicateAccount',
+  components: { Button },
+  data() {
+    return {
+      key: 'duplicate-account',
+    };
+  },
+  methods: {
+    goToLogin() {
+      this.$router.push({ path: '/auth/login' });
```
Comment (abdullahcalisir12) [inline src/components/inputs/DuplicateAccount.vue:55]: use name here instead of path.

File: src/components/inputs/DuplicateAccount.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      enter-to-class="opacity-100 scale-100"
+      leave-active-class="transition ease-in duration-150 transform"
+      leave-class="opacity-100 scale-100"
+      leave-to-class="opacity-0 scale-95"
+  >
+    <div
+      class="flex justify-center items-center fixed w-screen h-screen top-0 left-0 z-50 bg-black bg-opacity-38"
+    >
+      <div class="bg-secondary shadow-24dp p-4 rounded-lg">
+        <div v-if="key === 'duplicate-account'">
```
Comment (abdullahcalisir12) [inline src/components/inputs/DuplicateAccount.vue:14]: is there any possibility for key not being duplicate-account??? if no, then this key is unnecessary. Remove it.

File: src/components/forms/CustomerInfo.vue [hunk 3]
```diff
@@ -154,6 +161,12 @@ export default {
         })
         .catch(error => {
           this.registerError = error.response.data[0].description;
+          if (error.response.status === 400) {
+            if (error.response.data[0].code == "DuplicateUserName"){
```
Comment (abdullahcalisir12) [inline src/components/forms/CustomerInfo.vue:165]: no need to use nested if here

File: src/components/inputs/DuplicateAccount.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<template>
```
Comment (mort3za) [inline src/components/inputs/DuplicateAccount.vue:1]: Move this component under `src/components/` folder.

File: src/components/inputs/DuplicateAccount.vue (hunk 2 above)
Comment (joeelia) [reply src/components/inputs/DuplicateAccount.vue]: This has been resolved in latest commit

File: src/components/forms/CustomerInfo.vue (hunk 3 above)
Comment (joeelia) [reply src/components/forms/CustomerInfo.vue]: This has been resolved in latest commit

File: src/components/inputs/DuplicateAccount.vue (hunk 4 above)
Comment (joeelia) [reply src/components/inputs/DuplicateAccount.vue]: This has been resolved in latest commit

File: src/components/inputs/DuplicateAccount.vue (hunk 1 above)
Comment (joeelia) [reply src/components/inputs/DuplicateAccount.vue]: This has been resolved in latest commit

File: src/components/DuplicateAccount.vue [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+</template>
+
+<script>
+import Button from '@/components/Button.vue';
+export default {
+  name: 'DuplicateAccount',
+  components: { Button },
+  data() {
+    return {
+      key: 'duplicate-account',
```
Comment (abdullahcalisir12) [inline src/components/DuplicateAccount.vue:50]: key is not used anywhere. Better to remove

File: src/components/forms/CustomerInfo.vue [hunk 6]
```diff
@@ -154,6 +161,10 @@ export default {
         })
         .catch(error => {
           this.registerError = error.response.data[0].description;
+          if (error.response.data[0].code == "DuplicateUserName"){
```
Comment (abdullahcalisir12) [inline src/components/forms/CustomerInfo.vue:164]: Our backend is a bit strange it returns errors in different formats. So if data or data[0] are not exist app will crash. It's better to check if they are exist.

File: src/components/DuplicateAccount.vue (hunk 5 above)
Comment (abdullahcalisir12) [reply src/components/DuplicateAccount.vue:50]: @joeelia maybe you didn't see this comment so I tagged you.

File: src/components/forms/CustomerInfo.vue [hunk 7]
```diff
@@ -154,6 +161,12 @@ export default {
         })
         .catch(error => {
           this.registerError = error.response.data[0].description;
+          if (error){
+            if (error.response.data[0].code == "DuplicateUserName"){
```
Comment (abdullahcalisir12) [inline src/components/forms/CustomerInfo.vue:165]: I meant something like

`this.registerError = error.response.data[0] && error.response.data[0].description`

`if (error.response.data[0] && error.response.data[0].code === "DuplicateUserName")`

Comment (mort3za) [reply src/components/forms/CustomerInfo.vue:165]: This can cause errors. Can't merge because of this.  
I'll fix it If I found a time.

**Pull request merged**